service/block/fraud_test.go

```diff
@@ -0,0 +1,44 @@
+package block
+
+import (
+	"testing"
+
+	"github.com/celestiaorg/celestia-core/testutils"
+	"github.com/celestiaorg/celestia-node/service/header"
+	"github.com/stretchr/testify/require"
+)
+
+func Test_validateEncoding_Successful(t *testing.T) {
```

Comment:
General remark: I'd prefer table-driven tests and also a few non-happy-path cases always.

Reply:
Oh yes I prefer them too. I forgot to add the unhappy case.